subsys/bluetooth/host/iso.c

```diff
@@ -705,9 +705,41 @@ void bt_iso_recv(struct bt_conn *iso, struct net_buf *buf, uint8_t flags)
 #endif /* CONFIG_BT_ISO_UNICAST) || defined(CONFIG_BT_ISO_SYNC_RECEIVER */
 
 #if defined(CONFIG_BT_ISO_UNICAST) || defined(CONFIG_BT_ISO_BROADCASTER)
+static uint16_t iso_chan_max_data_len(const struct bt_iso_chan *chan,
+				      uint32_t ts)
+{
+	uint16_t max_data_len;
+
+	if (chan->qos->tx == NULL) {
+		return 0;
+	}
+
+	max_data_len = chan->qos->tx->sdu;
+
+	/* TODO: Remove when we support host ISO fragmentation */
+	if (ts == BT_ISO_TIMESTAMP_NONE) {
+		if (bt_dev.le.iso_mtu < BT_HCI_ISO_DATA_HDR_SIZE) {
+			max_data_len = 0;
+		} else {
+			max_data_len = MIN(bt_dev.le.iso_mtu - BT_HCI_ISO_DATA_HDR_SIZE,
+					   max_data_len);
+		}
+	} else {
+		if (bt_dev.le.iso_mtu < BT_HCI_ISO_TS_DATA_HDR_SIZE) {
+			max_data_len = 0;
+		} else {
+			max_data_len = MIN(bt_dev.le.iso_mtu - BT_HCI_ISO_TS_DATA_HDR_SIZE,
+					   max_data_len);
+		}
+	}
```

Comment:
```suggestion
	if (ts == BT_ISO_TIMESTAMP_NONE) {
		max_data_len = MIN(bt_dev.le.iso_mtu - BT_HCI_ISO_DATA_HDR_SIZE,
				   max_data_len);
	} else {
		max_data_len = MIN(bt_dev.le.iso_mtu - BT_HCI_ISO_TS_DATA_HDR_SIZE,
				   max_data_len);
	}
```

Reply:
Hmm, I see what you are trying here (i.e. move the `BT_HCI_ISO_DATA_HDR_SIZE` check to earlier), but we will need two checks for that, as `BT_HCI_ISO_DATA_HDR_SIZE` and `BT_HCI_ISO_TS_DATA_HDR_SIZE` aren't the same size. 

The suggestion you provide here, with https://github.com/zephyrproject-rtos/zephyr/pull/48068/files#r935272842, doesn't work as `bt_dev.le.iso_mtu - BT_HCI_ISO_TS_DATA_HDR_SIZE` may underflow.

Reply:
Updated after realizing that we actually do support fragmentation, so this part has been simplified.